vm/array.go

```diff
@@ -473,6 +489,70 @@ func builtinArrayInstanceMethods() []*BuiltInMethodObject {
 			},
 		},
 		{
+			// Returns a new array that is a one-dimensional flattening of self.
+			//
+			// ```ruby
+			// a = [ 1, 2, 3 ]
+			// b = [ 4, 5, 6, [7, 8] ]
+			// c = [ a, b, 9, 10 ] # => [[1, 2, 3], [4, 5, 6, [7, 8]], 9, 10]
+			// c.flatten # => [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			// ```
+			// @return [Array]
+			Name: "flatten",
+			Fn: func(receiver Object) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *callFrame) Object {
+					arr := receiver.(*ArrayObject)
+
+					if len(args) != 0 {
```

Comment:
I know ruby's `flatten` method can take an argument: level. But I thought I do not have to implement that feature right now (since I myself never ran across the situation I have to give level to flatten.).

If you think that is necessary, I'm happy to work on it! 👍